src/dgpost/utils/plot.py

```diff
@@ -0,0 +1,177 @@
+
+    Parameters
+    ----------
+    style
+        A dictionary object containing valid key/value pairs.
+
+    """
+    if style is not None:
+        matplotlib.rcParams.update(style)
+    else:
+        matplotlib.rcdefaults()
+
+
+def plt_axes(ax: matplotlib.axes.Axes, table: pd.DataFrame, ax_args: dict):
+    """
+    Processes ax_args and plots the data
+
+    Parameters
+    ----------
+    ax
+        axes to be plotted to
+    table
+        dataframe containing the data
+    ax_args
+        arguments for the axes
+
+    """
+    series: list[dict] = ax_args.pop("series", [])
+
+    for method, arguments in ax_args.pop("methods", {}).items():
+        attr = ax
+        # allows for method calls on attributes
+        for met in method.split("."):
+            attr = getattr(attr, met)
+        attr(**arguments)
+    ax.set(**ax_args)
+
+    for element in series:
+        # datas processing
+        # should I use here separate_data and pQ from dgpost.transform.helpers?
+        # but with that I will lose the access to the index for x_data
+        y = element.pop("y")
+        # check if multiple columns should be plotted
+        if y.endswith("*"):
+            elem_step = 0  # iterator step to keep track which property should be selected if list is given
+            for col in table.columns:
+                # check for appropriate columns names
+                if y[:-1] in col:
+                    elem = element.copy()  # copy specs -> maybe rename element to specs?
+                    for el, val in element.items():
+                        # check if any of the specs in elements ends with s and is a list
+                        if el.endswith("s") and isinstance(val, list):
+                            # select the property that corresponds to iterator step
+                            elem.pop(el)
+                            elem[el[:-1]] = val[elem_step % len(val)]
+                    elem["y"] = col  # set the y data name
+                    # append the new spec to the series and increase step by one
+                    series.append(elem)
+                    elem_step += 1
+            # don't plot anything for this element
+            # TODO: update doc with this new feature
+            continue
+
+        y_data = table[y]
+        y_values = unp.nominal_values(y_data.array)
+        y_err = unp.std_devs(y_data.array)
+        y_unit = table.attrs.get("units", {}).get(y, "")
+
+        if x := element.pop("x", None):
+            x_data = table[x]
+            x_values = unp.nominal_values(x_data)
+            x_err = unp.std_devs(x_data)
+            x_unit = table.attrs.get("units", {}).get(x, "")
+        else:
+            x_values = y_data.index
+            x_err = None
+            x_unit = None
+
+        kind = element.pop("kind", "scatter")
+
+        if "label" not in element:
+            element["label"] = y + (f"[{y_unit}]" if y_unit else "")
```

Comment:
This can be a single f-string:
```suggestion
            element["label"] = f"{y} [{y_unit}]" if y_unit else y
```